tests/apollo/test_skvbc_chaotic_startup.py

```diff
@@ -237,24 +218,136 @@ async def test_f_staggered_replicas_requesting_vc(self, bft_network, skvbc, nurs
 
         await trio.sleep(5)  # TODO: remove when bft_network.wait_for_view also waits for system liveness.
 
-        for r in bft_network.all_replicas(without={replica_to_stop}):
+        await self._verify_replicas_are_in_view(view, bft_network.all_replicas(without={replica_to_stop}), bft_network)
+
+        await self._verify_the_system_is_able_to_make_progress(bft_network, skvbc, expected_next_primary, bft_network.all_replicas(without={replica_to_stop}))
+
+    async def _test_f_minus_one_staggered_replicas_requesting_vc(self, bft_network, skvbc, nursery, with_catchup):
+        """
+        In this test we check that if f-1 replicas have started to run and initiated viewchange, then the system is
+        still able to make progress in both cases (1) when we leave the f-1 replicas to try and run the view change,
+        and (2) when we assume that the next primary may come from the f-1 replicas, and thus they have to catch up
+        with the rest 2f + 2 replicas.
+        For that we perform the following steps:
+        1. start f - 1 replicas
+        2. wait for those replicas to initiate view change
+        3. start the rest 2f + 2 replicas
+        4. make sure the system is able to make progress
+        5. stop the current primary
+        6. wait for the system to complete a view change
+        7. make sure the system is able to make progress
+        """
+        n = bft_network.config.n
+        f = bft_network.config.f
+        c = bft_network.config.c
+
+        initial_view = initial_primary = 0
+        expected_next_view = expected_next_primary = 1
+
+        # We start by choosing a random set of f - 1 replicas.
+        # If we are not suppose thus replicas to catch up, then both, the current primary and the next can't be choose
+        excluded = {initial_primary}
```

Comment:
make sure to also exclude the next primary :) otherwise you may end up with 2 consecutive view changes!

Reply:
If we wait for the other replicas to catch up we don't need to exclude the next primary, because it should catch up later

Reply:
I see, thanks!